### PR TITLE
No pthread

### PR DESCRIPTION
**Pull request by mgautierfr**
This PR remove strong dependency on pthread.
We use c++11 thread library which uses native thread and mutex api.

We still use pthread on linux but it allow use to don't use a wrapping pthread library on Windows.

### CODE REVIEW

File: src/writer/cluster.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ class Cluster {
     ClusterData _data;
     mutable Blob compressed_data;
     std::string tmp_filename;
-    mutable pthread_mutex_t m_closedMutex;
+    mutable std::mutex m_closedMutex;
     bool closed = false;
```
Comment (veloman-yunkan) [inline src/writer/cluster.h:92]: Isn't it better to use `std::atomic<bool>` instead (and get rid of the mutex)?

File: src/writer/creator.cpp [hunk 2]
```diff
@@ -88,12 +88,11 @@ namespace zim
...
       {
-        pthread_t thread;
-        pthread_create(&thread, NULL, taskRunner, this->data.get());
-        data->workerThreads.push_back(thread);
+        std::thread thread(taskRunner, this->data.get());
+        data->workerThreads.push_back(std::move(thread));
       }
 
-      pthread_create(&data->writerThread, NULL, clusterWriter, this->data.get());
+      data->writerThread = std::move(std::thread(clusterWriter, this->data.get()));
```
Comment (veloman-yunkan) [inline src/writer/creator.cpp:95]: I believe that `std::move()` is not required here.

File: meson.build [hunk 3]
```diff
@@ -45,15 +45,21 @@ conf.set('ENABLE_XAPIAN', xapian_dep.found())
...
     extra_cpp_args = ['-DSORTPP_PASS']
 else
-    thread_dep = dependency('threads')
     extra_link_args = []
     extra_cpp_args = []
 endif
+
+compiler = meson.get_compiler('cpp')
+if compiler.get_id() == 'gcc' and build_machine.system() == 'linux'
+  # C++ std::stread is implemented using pthread on linux by gcc
```
Comment (veloman-yunkan) [inline meson.build:57]: std::stread is a typo

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
A couple of minor suggestions for improvement (mainly as a proof that I actually reviewed this PR :smiley: )

**Comment by kelson42** [comment]
Needs to be rebased.

File: src/writer/cluster.h (hunk 1 above)
Comment (mgautierfr) [reply src/writer/cluster.h]: Yes, this is done.

File: src/writer/creator.cpp (hunk 2 above)
Comment (mgautierfr) [reply src/writer/creator.cpp]: Yes, I've removed it.

**Comment by kelson42** [comment]
@veloman-yunkan After latest @mgautierfr changes, does this branch can be merged?

**Comment by veloman-yunkan** [comment]
@kelson42 Yes. @mgautierfr didn't re-request review, so I wasn't sure if he was done with it.

**Pull request merged**